pkg/resolvers/k8s/resolver.go

```diff
@@ -108,10 +106,6 @@ func parseTarget(targetName string) (targetEntry, error) {
 		target.port = targetPort{
 			value: p,
 		}
-		if _, err := strconv.Atoi(p); err != nil {
-			// NaN
-			target.port.isNamed = true
-		}
```

Comment:
I wonder if this should be turned into a panic or an error since we've changed behaviour…

Reply:
@thommay That's a good question, this change is a bit scary indeed, better to be explicit, going to have a look.